### PR TITLE
[metasrv] fix drop table on unknown db test

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

With this fix, PR: #4175 should be passed.

## Changelog

- Improvement

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/meta/raft-store/src/state_machine/sm.rs [hunk 1]
```diff
@@ -848,10 +848,10 @@ impl StateMachine {
         let seq_dbi = txn_db_lookup
             .get(&(DatabaseLookupKey::new(tenant.to_string(), db_name.to_string())))?
             .ok_or_else(|| {
-                MetaStorageError::AppError(AppError::UnknownDatabase(UnknownDatabase::new(
+                AppError::UnknownDatabase(UnknownDatabase::new(
                     db_name.to_string(),
                     "txn_get_database_id".to_string(),
-                )))
+                ))
             })?;
```
Comment (drmingdrmer) [inline common/meta/raft-store/src/state_machine/sm.rs:855]: `From<AppError> for MetaStorageError` is already implemented. 
Thus IMHO this change has nothing to do with this issue. not?

File: metasrv/src/meta_service/meta_leader.rs [hunk 2]
```diff
@@ -190,7 +190,14 @@ impl<'a> MetaLeader<'a> {
...
 
         match write_rst {
-            Ok(resp) => Ok(resp.data),
+            Ok(resp) => {
+                let data = resp.data;
+                match data {
+                    AppliedState::AppError(ae) => Err(MetaError::from(ae)),
+                    _ => Ok(data),
+                }
+            }
```
Comment (drmingdrmer) [inline metasrv/src/meta_service/meta_leader.rs:199]: Nice catch!!!

**Review by drmingdrmer — APPROVED** [review]

**Comment by lichuang** [comment]
LGTM

**Pull request merged**